src/Compilers/Core/Desktop/SimpleAnalyzerAssemblyLoader.cs

```diff
@@ -0,0 +1,193 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
```

Comment:
Yes.  But we can abstract this away by having the type take a `Func<string, Assembly>`.  The instantiator can then provide the implementation based on whether it was compiled for Desktop or CoreCLR.